cub/test/catch2_test_device_run_length_encode_non_trivial_runs.cu

```diff
@@ -368,3 +388,98 @@ C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns does not run out of memory", "[d
   REQUIRE(out_lengths.front() == magic_number);
   REQUIRE(out_offsets.front() == magic_number);
 }
+
+C2H_TEST("DeviceRunLengthEncode::NonTrivialRuns works for a large number of items",
+         "[device][run_length_encode][skip-cs-initcheck][skip-cs-racecheck][skip-cs-synccheck][!mayfail]",
+         offset_types)
+try
+{
+  using offset_type     = typename c2h::get<0, TestType>;
+  using run_length_type = offset_type;
+
+  ::cuda::std::size_t random_range = GENERATE(take(1, random((1 << 20), (1 << 22))));
+  const auto num_items             = detail::make_large_offset<offset_type>(random_range);
+  CAPTURE(c2h::type_name<offset_type>(), c2h::type_name<run_length_type>(), num_items);
+
+  auto counting_it = cuda::make_counting_iterator(offset_type{0});
+
+  // Input iterator: repeat odd numbers once and even numbers twice
+  auto input_item_it = cuda::make_transform_iterator(counting_it, index_to_item_op{});
+  // Number of non-trivial runs is number of full(!) three-item (i.e., even, odd, odd) groups
+  const auto num_uniques = num_items / 3;
+
+  // Prepare helper to check the unique items being written: we expect the i-th item corresponding to value i
+  auto check_offset_out_helper = detail::large_problem_test_helper(num_uniques);
+  auto expected_offsets_it     = cuda::make_transform_iterator(counting_it, run_index_to_offset_op{});
+  auto check_offset_out_it     = check_offset_out_helper.get_flagging_output_iterator(expected_offsets_it);
+
+  // Prepare helper to check the run-lengths being written: i-th item corresponding to value i
+  // We repeat each number once for the first num_small_runs number of items and all subsequent numbers twice
```

Comment:
Is this comment correct? I don't understand what `num_small_runs` means here. Other tests do have such a variable.